pyvcloud/vcd/system.py

```diff
@@ -57,13 +59,64 @@ def delete_org(self, org_name, force=None, recursive=None):
         """
         Delete an organization.
         :param org_name: (str): name of the org to be deleted.
-        :param force: (bool): pass force=True  along with recursive=True to 
-        remove an organization and any objects it contains, regardless of 
+        :param force: (bool): pass force=True  along with recursive=True to
+        remove an organization and any objects it contains, regardless of
         their state.
-        :param recursive: (bool): pass recursive=True  to remove an 
-        organization and any objects it contains that are in a state that 
+        :param recursive: (bool): pass recursive=True  to remove an
+        organization and any objects it contains that are in a state that
         normally allows removal.
         """  # NOQA
         org = self.client.get_org_by_name(org_name)
         org_href = get_admin_href(org.get('href'))
         return self.client.delete_resource(org_href, force, recursive)
+
+    def list_provider_vdcs(self):
+        if self.admin_resource is None:
+            self.admin_resource = self.client.get_resource(self.admin_href)
+        if hasattr(self.admin_resource.ProviderVdcReferences,
+                   'ProviderVdcReference'):
+            return self.admin_resource.ProviderVdcReferences.\
+                        ProviderVdcReference
+        else:
+            return []
+
+    def get_provider_vdc(self, pvdc_name):
+        for pvdc in self.list_provider_vdcs():
+            if pvdc.get('name') == pvdc_name:
+                return pvdc
+        raise Exception('PVDC not found (or)'
+                        ' Access to resource is forbidden')
+
+    def list_provider_vdc_storage_profiles(self, name=None):
+        if name is not None:
+            query_filter = 'name==%s' % name
+        else:
+            query_filter = None
+        q = self.client.get_typed_query(
+            'providerVdcStorageProfile',
+            query_result_format=QueryResultFormat.RECORDS,
+            qfilter=query_filter)
+        return list(q.execute())
+
+    def get_provider_vdc_storage_profile(self, name):
+        for profile in self.list_provider_vdc_storage_profiles(name):
+            if profile.get('name') == name:
+                return profile
+        raise Exception('PVDCSP not found (or)'
```

Comment:
pVdc_storage profile not found? (or) mentioning the SP name in the error msg helps.

Reply:
PVDCSP is not an acronym commonly used.

Reply:
done